tools/check-go-version-dockerfile.sh

```diff
@@ -28,8 +28,17 @@ target_go_version="$1"
 # Search for Dockerfiles in the current directory and its subdirectories
 dockerfiles=$(find . -type f -name "*.Dockerfile" -o -name "Dockerfile")
 
+# Exclude non-project files.
+exclusions=(./vendor)
```

Comment:
Since we're doing a `find` above, we should use `-not -path` there (same for the YAML): https://stackoverflow.com/questions/4210042/how-do-i-exclude-a-directory-when-using-find

Reply:
I didn't go down that route because I thought this would crowd the find command if we were to add more elements to the exclusion list. Here are two more ways to do this, but I think both ways are more complicated than what we've got committed right now:

1. 
```
exclude_dirs=(./vendor ./another_directory)
exclusion_pattern=""

for dir in "${exclude_dirs[@]}"; do
    exclusion_pattern+=" -path $dir -prune -o"
done

dockerfiles=$(find . $exclusion_pattern -type f \( -name "*.Dockerfile" -o -name "Dockerfile" \) -print)
```

2.
```
exclusions=(./vendor ./another_directory)
exclusion_params=""

for dir in "${exclusions[@]}"; do
    exclusion_params+=" -e $dir"
done

dockerfiles=$(find . -type f \( -name "*.Dockerfile" -o -name "Dockerfile" \) | grep -v $exclusion_params)
```

Reply:
I would go with option 1 then.
Although just to avoid the somewhat tricky syntax of bash lists, we could also just do:
```
dockerfiles=$(find . \
  -path ./vendor -prune -o \
  -path ./another_directory -o \
  -type f \
  \(-name "*.Dockerfile" -o -name "Dockerfile" \) \
  -print\
)
```

Reply:
I've gone down the route that you suggest in your last comment ^